File: src/main/java/com/dashboard/security/CustomEmployeeDetailsService.java
package dashboard.security;

import dashboard.exceptions.EmployeeNotFoundException;
import dashboard.models.jpa.Employee;
import dashboard.models.security.CustomEmployeeDetails;
import dashboard.repositories.EmployeeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class CustomEmployeeDetailsService implements UserDetailsService {

    @Autowired
    private EmployeeRepository employeeRepository;

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        Optional<Employee> employee = employeeRepository.findByUsername(username);
        employee.orElseThrow(EmployeeNotFoundException::new);
        return employee.map(CustomEmployeeDetails::new).get();
    }
}

class UserService {

    @Autowired
    PasswordEncoder passwordEncoder;
    @Autowired
    EmployeeRepository employeeRepository;

    public long saveUser(Employee employee) {
        Employee savedEntity = employeeRepository.save(employee);
        return savedEntity.getId();
    }

    public Employee findUserByUsername(String userName) {
        return employeeRepository.findByUsername(userName).orElse(null);
    }
}

File: src/main/kotlin/com/employeeDashboard/security/CustomEmployeeDetailsService.kt
//package com.employeeDashboard.security
//
//import com.dashboard.exception.EmployeeNotFoundException
//import com.dashboard.model.Employee
//import com.dashboard.model.security.CustomEmployeeDetails
//import com.dashboard.repository.EmployeeRepository
//import org.springframework.beans.factory.annotation.Autowired
//import org.springframework.security.core.userdetails.UserDetails
//import org.springframework.security.core.userdetails.UserDetailsService
//import org.springframework.security.core.userdetails.UsernameNotFoundException
//import org.springframework.security.crypto.password.PasswordEncoder
//import org.springframework.stereotype.Service
//import java.util.*
//import kotlin.Throws
//
//@Service
//class CustomEmployeeDetailsService : UserDetailsService {
//    @Autowired
//    private val employeeRepository: EmployeeRepository? = null
//    @Throws(UsernameNotFoundException::class)
//    fun loadUserByUsername(username: String?): UserDetails {
//        val employee: Optional<Employee> = employeeRepository.findByUsername(username)
//        employee.orElseThrow<RuntimeException> { EmployeeNotFoundException() }
//        return employee.map<Any> { CustomEmployeeDetails() }.get()
//    }
//}
//
//internal class UserService {
//    @Autowired
//    var passwordEncoder: PasswordEncoder? = null
//
//    @Autowired
//    var employeeRepository: EmployeeRepository? = null
//    fun saveUser(employee: Employee?): Long {
//        val savedEntity: Employee = employeeRepository.save(employee)
//        return savedEntity.getId()
//    }
//
//    fun findUserByUsername(userName: String?): Employee {
//        return employeeRepository.findByUsername(userName).orElse(null)
//    }
//}